### ISSUE TITLE
AdsRDD - redundant RecordMovement calls 

### ISSUE THREAD

**Issue by christianschmid200272** [X-Sharp/XSharpPublic#1395]
Hello,

in the class AdsRDD the method GoCold() was introduced with Commit https://github.com/X-Sharp/XSharpPublic/commit/447e79a8a470c9cbb010596d662c1f6bc359b62d, GoCold() returns the result of RecordMovement() at the end, Append(), Skip() and GoTo() now first call GoCold() and then RecordMovement again. RecordMovement itself executes RecordMovement for each relation. 

Since I am searching for the problem what triggers the negative runtime behavior with bBrowser:Server:Notify(NOTIFYFILECHANGE) in connection with servers with many relations, we came across this change. I cannot say for sure whether this change really triggers the problem, the redundant calls are probably not optimal. For a test, we reverted to the last XSharp runtime before this change and the runtime is good again.

Tested with Runtime 2.16.0.5, performance is ok.
The current runtime used is 2.18.0.4, so the performance is not ok.

An example application is difficult to create.

**Comment by RobertvanderHulst** [comment]
I think can change GoCold to only call RecordMovement when the file was "Hot". That should fix the issue I think.
I am calling RecordMovement here because a change in the contents of the file could result in a filter condition or index condition change, which could result in an EOF flag to be changed.

**Comment by RobertvanderHulst** [comment]
Chris, can you create a test RDD and send that to Christian?

**Comment by cpyrgas** [comment]
Sure, here's an update to XSharp.RDD, but also to XSharp.Core and XSharp.RT, as there might be inter dependencies to other recent updates. Christian, you need to update all 3 files in the GAC in the respective folders under

C:\Windows\Microsoft.NET\assembly\GAC_MSIL\

(and of course X# build 2.18 needs to be already installed)

https://www.dropbox.com/scl/fi/xeto4gqy5xe433xf6j3cs/RuntimeUpdate_2020_11_30.zip?rlkey=n9zm582l8flb6ubjoo7zm52xx&dl=0

**Comment by christianschmid200272** [comment]
Hello,
unfortunately I'm only getting in touch now, but here in southern Germany we are currently drowning in snow and have therefore been shoveling snow all weekend.
I have tested the change and the notifications are significantly reduced. Unfortunately, the hoped-for improvement in runtime has not materialized, so I will create another problem case in GitHub today because I was able to identify a specific point where a significantly longer runtime occurs after an update from X# versions 2.17.03 to 2.18.04.
